// chart-1$ajv-keywords@3.5.2/keywords/index.js
'use strict';

Liferay.Loader.define('chart-1$ajv-keywords@3.5.2/keywords/index', ['module', 'exports', 'require', './instanceof', './range', './regexp', './typeof', './dynamicDefaults', './allRequired', './anyRequired', './oneRequired', './prohibited', './uniqueItemProperties', './deepProperties', './deepRequired', './formatMinimum', './formatMaximum', './patternRequired', './switch', './select', './transform'], function (module, exports, require) {
  var define = undefined;
  var global = window;
  {
    module.exports = {
      'instanceof': require('./instanceof'),
      range: require('./range'),
      regexp: require('./regexp'),
      'typeof': require('./typeof'),
      dynamicDefaults: require('./dynamicDefaults'),
      allRequired: require('./allRequired'),
      anyRequired: require('./anyRequired'),
      oneRequired: require('./oneRequired'),
      prohibited: require('./prohibited'),
      uniqueItemProperties: require('./uniqueItemProperties'),
      deepProperties: require('./deepProperties'),
      deepRequired: require('./deepRequired'),
      formatMinimum: require('./formatMinimum'),
      formatMaximum: require('./formatMaximum'),
      patternRequired: require('./patternRequired'),
      'switch': require('./switch'),
      select: require('./select'),
      transform: require('./transform')
    };
  }
});
