// chart-1$ajv-keywords@3.5.2/keywords/regexp.js
'use strict';

Liferay.Loader.define('chart-1$ajv-keywords@3.5.2/keywords/regexp', ['module', 'exports', 'require'], function (module, exports, require) {
  var define = undefined;
  var global = window;
  {
    module.exports = function defFunc(ajv) {
      defFunc.definition = {
        type: 'string',
        inline: function (it, keyword, schema) {
          return function getRegExp() {
            try {
              if (typeof schema == 'object') return new RegExp(schema.pattern, schema.flags);

              var rx = schema.match(/^\/(.*)\/([gimuy]*)$/);
              if (rx) return new RegExp(rx[1], rx[2]);
              throw new Error('cannot parse string into RegExp');
            } catch (e) {
              console.error('regular expression', schema, 'is invalid');
              throw e;
            }
          }() + '.test(data' + (it.dataLevel || '') + ')';
        },
        metaSchema: {
          type: ['string', 'object'],
          properties: {
            pattern: { type: 'string' },
            flags: { type: 'string' }
          },
          required: ['pattern'],
          additionalProperties: false
        }
      };

      ajv.addKeyword('regexp', defFunc.definition);
      return ajv;
    };
  }
});
