// chart-1$ajv-keywords@3.5.2/keywords/switch.js
'use strict';

Liferay.Loader.define('chart-1$ajv-keywords@3.5.2/keywords/switch', ['module', 'exports', 'require', './_util', './dotjs/switch'], function (module, exports, require) {
  var define = undefined;
  var global = window;
  {
    var util = require('./_util');

    module.exports = function defFunc(ajv) {
      if (ajv.RULES.keywords.switch && ajv.RULES.keywords.if) return;

      var metaSchemaRef = util.metaSchemaRef(ajv);

      defFunc.definition = {
        inline: require('./dotjs/switch'),
        statements: true,
        errors: 'full',
        metaSchema: {
          type: 'array',
          items: {
            required: ['then'],
            properties: {
              'if': metaSchemaRef,
              'then': {
                anyOf: [{ type: 'boolean' }, metaSchemaRef]
              },
              'continue': { type: 'boolean' }
            },
            additionalProperties: false,
            dependencies: {
              'continue': ['if']
            }
          }
        }
      };

      ajv.addKeyword('switch', defFunc.definition);
      return ajv;
    };
  }
});
