// chart-1$ajv-keywords@3.5.2/keywords/typeof.js
'use strict';

Liferay.Loader.define('chart-1$ajv-keywords@3.5.2/keywords/typeof', ['module', 'exports', 'require'], function (module, exports, require) {
  var define = undefined;
  var global = window;
  {
    var KNOWN_TYPES = ['undefined', 'string', 'number', 'object', 'function', 'boolean', 'symbol'];

    module.exports = function defFunc(ajv) {
      defFunc.definition = {
        inline: function (it, keyword, schema) {
          var data = 'data' + (it.dataLevel || '');
          if (typeof schema == 'string') return 'typeof ' + data + ' == "' + schema + '"';
          schema = 'validate.schema' + it.schemaPath + '.' + keyword;
          return schema + '.indexOf(typeof ' + data + ') >= 0';
        },
        metaSchema: {
          anyOf: [{
            type: 'string',
            enum: KNOWN_TYPES
          }, {
            type: 'array',
            items: {
              type: 'string',
              enum: KNOWN_TYPES
            }
          }]
        }
      };

      ajv.addKeyword('typeof', defFunc.definition);
      return ajv;
    };
  }
});
