// chart-1$ajv@6.12.6/lib/definition_schema.js
'use strict';

Liferay.Loader.define('chart-1$ajv@6.12.6/lib/definition_schema', ['module', 'exports', 'require', './refs/json-schema-draft-07.json'], function (module, exports, require) {
  var define = undefined;
  var global = window;
  {
    var metaSchema = require('./refs/json-schema-draft-07.json');

    module.exports = {
      $id: 'https://github.com/ajv-validator/ajv/blob/master/lib/definition_schema.js',
      definitions: {
        simpleTypes: metaSchema.definitions.simpleTypes
      },
      type: 'object',
      dependencies: {
        schema: ['validate'],
        $data: ['validate'],
        statements: ['inline'],
        valid: { not: { required: ['macro'] } }
      },
      properties: {
        type: metaSchema.properties.type,
        schema: { type: 'boolean' },
        statements: { type: 'boolean' },
        dependencies: {
          type: 'array',
          items: { type: 'string' }
        },
        metaSchema: { type: 'object' },
        modifying: { type: 'boolean' },
        valid: { type: 'boolean' },
        $data: { type: 'boolean' },
        async: { type: 'boolean' },
        errors: {
          anyOf: [{ type: 'boolean' }, { const: 'full' }]
        }
      }
    };
  }
});
